src/tdymemory.c

```diff
@@ -236,57 +236,57 @@ PetscErrorCode TDyDeallocate_RealArray_4D(PetscReal ****array_4D, PetscInt ndim_
 }
 
 /* ---------------------------------------------------------------- */
```

Comment:
It doesn't seem like these allocation functions are buying us a lot--we could just use [PETScMalloc](https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Sys/PetscMalloc.html#PetscMalloc) instead, which has (essentially) the same two arguments.